src/main/java/io/cryostat/MainModule.java

```diff
@@ -149,6 +151,13 @@ JMXServiceURL.class, new GsonJmxServiceUrlAdapter(logger))
         return builder.create();
     }
 
+    @Provides
+    @Singleton
+    public static ObjectMapper provideObjectMapper() {
+        return new ObjectMapper()
```

Comment:
I'm still not clear on why we need Jackson's `ObjectMapper`. What is this doing that the `gson` instance we're already using elsewhere doesn't/can't?

Reply:
I don't think gson is able to parse Java Maps (unless I'm wrong), which are the underlying type of the Notifications (or at least the one we care about) into POJOs. 
```
    public void writeMessage(Notification message) {
        notificationSource.notifyListeners(message);
        String json = gson.toJson(message);
         ...
    }
```
That being said, I could change the type of 
`notifyListeners` to take in json `String` instead of `Notification` and change `writeMessage`

to
```
        String json = gson.toJson(message);
        notificationSource.notifyListeners(json);
```
but that sort of gets rid of the generic `T` if I just pass json String.

But then we use gson to parse that into the POJOs if that's better.

Reply:
I'm quite sure Gson handles Maps.

`Map<X, Y> map = gson.fromJson(jsonString, new TypeToken<Map<X, Y> {}.getType())`

should do it.

But why is this needed exactly? You're performing a deserialization of a Notification's `.message()` content from JSON into a Java Map. How/where/when did the Notification's `Map` message become JSON to begin with?

Can we just avoid making it JSON to begin with here? It should only need to be made into JSON when we're going to write it out to the WebSocket connection and broadcast it over the network. This `NotificationListener` system is just passing object references within our same Java process so JSON serialization should not be needed or used.

Reply:
That makes sense, sorry about that.

Reply:
Wait, sorry it's a lot of questions but, 

> 
> to
> 
> ```
>         String json = gson.toJson(message);
>         notificationSource.notifyListeners(json);
> ```
this String json was already in the code being passed through the socket, I wasn't using it. I was just saying that to use `Gson` to parse the `Notification<T>`, I have to somehow use something to turn the `T` into something readable and that's where the jackson `ObjectMapper` came in, which doesn't have the extra step of converting the `T` into a String first, and then converting that String back into an `ActiveRecordingStopMatcher` object. It just converts the `T`, which is implicitly assumed to be a Java `Map<String, Object>>, into that resulting object.

Sorry, I realize that my wording was confusing, I wasn't actually using json at all.

Reply:
Oh, I'm not really using Jackson to do unchecked type casting, I'm just using it to easily convert the Map `notification.getMessage()` into something more readable which is the object
```
    private static class ActiveRecordingStopMatcher {
        private String target;
        private String name;
...
```
because otherwise, if I didn't use some sort of thing like Jackson, in 
```
    @Override
    public void callback(Notification<Map<String, Object>> notification) {
        String category = notification.getCategory();
```
I have to do 
`String targetId = notification.getMessage().get("targetId");`
`Object recording = notification.getMessage().get("recording");`
`String recordingName = ((Map<String, String>)recording).get("name"); \\Some way to cast Object into Map anyways
`invalidateCacheEntry(new ConnectionDescriptor(targetId), recordingName);`
`etc...`

Reply:
To me, it makes more sense to convert the `Map` into an object which is easily readable using Jackson annotations and `ObjectMapper`, and I don't think there's any other way around it since `Notification.getMessage()` in this case will always be of type `Map<String, Object>` otherwise, I have to do the manual Map field getting, like above.

Reply:
> Notification.getMessage() in this case will always be of type Map<String, Object>

"this case" meaning for this cache's listener, right? Other listeners on other notification categories could be expecting different notification message formats.

I'm just not quite convinced it's worth tying this implementation to the Jackson dependency just to avoid the "manual Map field getting" and the explicit cast from Object to Map. That's pretty standard and conventional, and portable if/when we migrate to another framework that may not already include Jackson as a dependency.

Reply:
Okay, sounds good, makes sense to me thanks for the feedback. ^^